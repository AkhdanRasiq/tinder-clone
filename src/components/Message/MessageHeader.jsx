import React from 'react'

import { Avatar } from '@mui/material'
import { styled } from '@mui/material/styles'
import Badge from '@mui/material/Badge'

const StyledBadgeOnline = styled(Badge)(({ theme }) => ({
  '& .MuiBadge-badge': {
    backgroundColor: '#44b700',
    color: '#44b700',
    width: '10px',
    height: '10px',
    borderRadius: '100%',
    right: 10,
    bottom: 15,
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
    '&::after': {
      position: 'absolute',
      top: -1,
      left: -1,
      width: '100%',
      height: '100%',
      borderRadius: '100%',
      animation: 'ripple 1.2s infinite ease-in-out',
      border: '1px solid currentColor',
      content: '""',
    },
  },
  '@keyframes ripple': {
    '0%': {
      transform: 'scale(.8)',
      opacity: 1,
    },
    '100%': {
      transform: 'scale(2.4)',
      opacity: 0,
    },
  },
}));

const StyledBadgeOffline = styled(Badge)(({ theme }) => ({
  '& .MuiBadge-badge': {
    backgroundColor: '#b70000',
    color: '#b70000',
    width: '10px',
    height: '10px',
    borderRadius: '100%',
    right: 10,
    bottom: 15,
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`
  }
}));

class MessageHeader extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isOnline: false
    }
  }

  componentDidMount() {
    window.addEventListener("authorOnline", this.handleEventStatus);
  }

  componentWillUnmount() {
    window.removeEventListener("authorOnline", this.handleEventStatus);
  }

  handleEventStatus = (event) => {
    this.setState({ isOnline: event.detail.isOnline })
  }

  render() {
    return (
      <div className='messageHeaderContainer'>
        {this.state.isOnline ?
          <StyledBadgeOnline
            overlap="circular"
            anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
            variant="dot"
          >
            <Avatar className='messageHeaderAvatar' alt="Avatar Image" src="https://www.logo.wine/a/logo/Tinder_(app)/Tinder_(app)-Flame-Logo.wine.svg" />
          </StyledBadgeOnline>
        :
          <StyledBadgeOffline
            overlap="circular"
            anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
            variant="dot"
          >
            <Avatar className='messageHeaderAvatar' alt="Avatar Image" src="https://www.logo.wine/a/logo/Tinder_(app)/Tinder_(app)-Flame-Logo.wine.svg" />
          </StyledBadgeOffline>
        }
      </div>
    )
  }
}

export default MessageHeader
