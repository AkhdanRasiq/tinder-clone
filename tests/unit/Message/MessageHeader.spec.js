import React from 'react'
import { shallow } from "enzyme"

import MessageHeader from "../../../src/components/Message/MessageHeader"

describe("MessageHeader.jsx", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <MessageHeader />
    )
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("renders user online", () => {
    wrapper.setState({ isOnline: true})

    expect(wrapper.state('isOnline')).toEqual(true)
    expect(wrapper.find('#userOnline').length).toBe(1)
  })

  it("renders user offline", () => {
    wrapper.setState({ isOnline: false})

    expect(wrapper.state('isOnline')).toEqual(false)
    expect(wrapper.find('#userOffline').length).toBe(1)
  })
})
