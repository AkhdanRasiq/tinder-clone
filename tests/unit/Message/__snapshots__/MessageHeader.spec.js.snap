// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageHeader.jsx renders properly 1`] = `ShallowWrapper {}`;
